libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -172,6 +172,19 @@ private List<VcStoreItem> getSuccessfulVCStoreItems(List<VcStoreItem> vcStoreIte
         return successfulVCStoreItems;
     }
 
+    public Optional<Boolean> getVCSuccessStatus(String userId, String criIssuer)
+            throws ParseException {
+        List<VcStoreItem> vcStoreItems = getVcStoreItems(userId);
```

Comment:
We have a method in this class, `getVcStoreItem(String userId, String criId)`, which will get the VC for the CRI for the user. This would mean you wouldn't have to pull them all, deserialize them and filter them.